improver/calibration/simple_bias_correction.py

```diff
@@ -0,0 +1,300 @@
+    """
+
+    def __init__(self):
+        """
+        Initialise class for applying simple bias correction.
+        """
+        self.correction_method = apply_additive_correction
+
+    def _get_mean_bias(self, bias_values: CubeList) -> Cube:
+        """
+        Evaluate the mean bias from the input cube(s) in bias_values.
+
+        Where multiple cubes are provided, each bias value must represent
+        a single forecast_reference_time to ensure that the resultant value
+        is the true mean over the set of reference forecasts. This is done
+        by checking the forecast_reference_time bounds; if a bias_value is
+        defined over a range of frt values (ie. bounds exist) an error will
+        be raised.
+
+        Args:
+            bias_values:
+                Cubelist containing the input bias cube(s).
+
+        Returns:
+            Cube containing the mean bias evaulated from set of bias_values.
+        """
+        # Currently only support for cases where the input bias_values are defined
+        # over a single forecast_reference_time.
+        if len(bias_values) == 1:
+            return bias_values[0]
+        else:
+            # Loop over bias_values and check bounds on each cube.
+            for bias_cube in bias_values:
+                if bias_cube.coord("forecast_reference_time").bounds is not None:
+                    raise ValueError(
+                        "Collapsing multiple bias values to a mean value is unsupported for "
+                        "bias values defined over multiple reference forecast values."
+                    )
+            bias_values = bias_values.merge_cube()
+            mean_bias = collapsed(
+                bias_values, "forecast_reference_time", iris.analysis.MEAN
+            )
+            return mean_bias
+
+    def process(
+        self,
+        forecast: Cube,
+        bias: CubeList,
+        lower_bound: Optional[float],
+        fill_masked_bias_values: Optional[bool] = False,
+    ) -> Cube:
+        """
+        Apply bias correction using the specified bias values.
+
+        Where a lower bound is specified, all values that fall below this
+        lower bound (after bias correction) will be remapped to this value
+        to ensure physically realistic values.
+
+        Args:
+            forecast:
+                The cube to which bias correction is to be applied.
+            bias:
+                The cubelist containing the bias values for which to use in
+                the bias correction.
+            lower_bound:
+                A lower bound below which all values will be remapped to
+                after the bias correction step.
+
+        Returns:
+            Bias corrected forecast cube.
+        """
+        bias = self._get_mean_bias(bias)
+
+        corrected_forecast = forecast.copy()
+        corrected_forecast.data = self.correction_method(
+            forecast, bias, fill_masked_bias_values
+        )
```

Comment:
Just wondering, if there's any concern about applying the bias to inappropriate forecasts? At the moment, there's no checking of e.g. forecast_reference_time or forecast_period, so any bias could be applied to any forecast. Is that flexibility intentional or an oversight?

Reply:
I've add a function to check the inputs prior to doing the bias-correction. This function checks that all bias datasets are consistent (where multiple are passed in) and that the forecast and bias data are consistent. Where either the `forecast_reference_time` valid-hour or `forecast_period` are different across datasets, a `ValueError` is raised.

Reply:
Is there a case where we might want to combine different forecast reference times? E.g. if we have only a short period of data and want to increase the sample size, and we think the bias is mostly due to location features and would be consistent over different forecast reference times. If we may want to use it this way, we could raise a warning instead of error.

Reply:
The flag sounds like a reasonable solution to me.